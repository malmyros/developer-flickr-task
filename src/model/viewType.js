module.exports = {
  LARGE: 'LARGE',
  THUMBNAIL: 'THUMBNAIL',
};
