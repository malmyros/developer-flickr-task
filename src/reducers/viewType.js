import { THUMBNAIL } from '../model/viewType';
import { CHANGE_VIEW_TYPE } from '../types';

const viewType = (state = THUMBNAIL, action) => {
  switch (action.type) {
    case CHANGE_VIEW_TYPE:
      return action.viewType;
    default:
      return state;
  }
};

export default viewType;
