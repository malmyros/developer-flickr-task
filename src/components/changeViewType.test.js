import React from 'react';
import { shallow } from 'enzyme';
import ChangeViewType from './changeViewType';
import { THUMBNAIL } from '../model/viewType';

describe('change view type', () => {
  it('should render without crashing', () => {
    const mockedCallback = jest.fn();
    const component = shallow(<ChangeViewType viewType={THUMBNAIL}
                                              onChangeViewType={mockedCallback}/>);
    expect(component).toMatchSnapshot();
  });
});
