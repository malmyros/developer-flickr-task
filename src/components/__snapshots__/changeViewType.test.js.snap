// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change view type should render without crashing 1`] = `
<div
  className="change-view-type-config"
>
  <ul
    className="view-types"
  >
    <li>
      <a
        className="active"
        onClick={[Function]}
        role="button"
        title="Thumbnail view"
      >
        <i
          className="fas fa-th-large"
        />
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        role="button"
        title="Large view"
      >
        <i
          className="fas fa-stop"
        />
      </a>
    </li>
  </ul>
</div>
`;
